ClangSharp.PInvokeGenerator/Cursors/Decls/Decl.cs

```diff
@@ -158,16 +158,33 @@ public static new Decl Create(CXCursor handle, Cursor parent)
             }
         }
 
-        private readonly Lazy<Type> _type;
+        private readonly Lazy<Cursor> _definition;
 
         protected Decl(CXCursor handle, Cursor parent) : base(handle, parent)
         {
             Debug.Assert(handle.IsDeclaration);
-            _type = new Lazy<Type>(() => TranslationUnit.GetOrCreateType(Handle.Type, () => Type.Create(Handle.Type, TranslationUnit)));
+
+            _definition = new Lazy<Cursor>(() => {
+                var cursor = TranslationUnit.GetOrCreateCursor(Handle.Definition, () => Create(Handle.Definition, this));
+                cursor.Visit(clientData: default);
+                return cursor;
+            });
         }
 
-        public bool IsAnonymous => Handle.IsAnonymous;
+        public CX_CXXAccessSpecifier AccessSpecifier => Handle.CXXAccessSpecifier;
```

Comment:
I've started, but not completed exposing some of the `CXCursor` properties where they are actually relevant. For example, `CXCursor.CXXAccessSpecifier` only returns useful information for declarations and `CXXBaseSpecifier`, everything else just returns `CX_CXXInvalidAccessSpecifier`

Reply:
I think I've completed exposing these properties for declarations now.